src/components/hooks/use-handle-lazy-load.tsx

```diff
@@ -0,0 +1,40 @@
+import React, { ReactNode, useEffect, useState } from 'react'
+type templateProps = {
+    lazytemplates?: ReactNode
+    targetId?: string
+    options?: object
+}
+const UseHandleLazyLoad = ({ lazytemplates, targetId, options }: templateProps) => {
+    const [isVisible, setIsVisible] = useState(false)
+    const [is_large_screen, setLargeScreen] = useState(false)
+
+    const handleIntersect = (entries, observer) => {
+        const [entry] = entries
+        if (typeof entry.isVisible === 'undefined') {
+            // The browser doesn't support Intersection Observer v2, falling back to v1 behavior.
+            setIsVisible(entry.isIntersecting)
+            observer.disconnect()
+            return
+        }
+        if (entry.isIntersecting) {
+            setIsVisible(entry.isIntersecting)
+            observer.disconnect()
+        }
+    }
+    useEffect(() => {
+        createObserver()
+        if (window.innerHeight >= 895) {
+            setLargeScreen(true)
+        }
+    }, [])
```

Comment:
this empty dependency array throws error on the console. 
you need to add createObserver to it
`}, [createObserver])`

Reply:
I guess it should be fine ,since it is supposed to execute only once after component render.
If is there any VP height changes, interceptor will handle that.